### PR TITLE
DT-442 Migration to Core22/Gnome42

### PR DESCRIPTION
**Pull request by sergio-costas**
# Pull Request Template

## Description

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

Fixes # (issue)

## Type of change

Please check only the options that are relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration.

**Test Configuration**:

* OS (please include version):
* Any other relevant environment information:

## Checklist:

- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have checked my code and corrected any misspellings

### CODE REVIEW

File: patches/patch1.diff [hunk 1]
```diff
@@ -0,0 +1,3160 @@
+diff --git a/data/meson.build b/data/meson.build
```
Comment (hellsworth) [inline patches/patch1.diff]: Please add a comment section to the top of the diff saying what this patch is for.

File: snapcraft.yaml [hunk 2]
```diff
@@ -18,39 +18,38 @@ slots:
 
 apps:
   gnome-2048:
-    extensions: [gnome-3-38]
+    extensions: [gnome]
     command: usr/bin/gnome-2048
     desktop: usr/share/applications/org.gnome.TwentyFortyEight.desktop
     common-id: org.gnome.TwentyFortyEight
+    environment:
+      DISABLE_WAYLAND: 'true'
```
Comment (hellsworth) [inline snapcraft.yaml:28]: Oof, disable wayland just forever? maybe add a comment guiding us when we can remove this.

File: snapcraft.yaml [hunk 3]
```diff
@@ -18,39 +18,38 @@ slots:
...
+    meson-parameters:
+      - --prefix=/usr
+      - --buildtype=release
     parse-info: [usr/share/metainfo/org.gnome.TwentyFortyEight.appdata.xml]
     override-pull: |
-      snapcraftctl pull
-      snapcraftctl set-version $(git describe --tags --abbrev=10)
+      craftctl default
+      craftctl set version=$(git describe --tags --abbrev=10)
+      patch -p1 < $CRAFT_PROJECT_DIR/patches/patch1.diff
```
Comment (hellsworth) [inline snapcraft.yaml]: Please rename the patch to something more intuitive and use *.diff

**Review by hellsworth — CHANGES_REQUESTED** [review]

File: patches/patch1.diff (hunk 1 above)
Comment (sergio-costas) [reply patches/patch1.diff]: Done.

File: snapcraft.yaml (hunk 3 above)
Comment (sergio-costas) [reply snapcraft.yaml]: Done.

File: snapcraft.yaml (hunk 2 above)
Comment (sergio-costas) [reply snapcraft.yaml:28]: Added a note specifying to try again when it gets ported from Clutter to Gtk4.

**Review by hellsworth — APPROVED** [review]
Looks great thanks!

**Pull request merged**